src/Compilers/VisualBasic/Portable/Compilation/MethodCompiler.vb

```diff
@@ -1384,12 +1405,16 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
             Dim allowOmissionOfConditionalCalls = _moduleBeingBuiltOpt Is Nothing OrElse _moduleBeingBuiltOpt.AllowOmissionOfConditionalCalls
             Dim lambdaDebugInfoBuilder = ArrayBuilder(Of LambdaDebugInfo).GetInstance()
             Dim closureDebugInfoBuilder = ArrayBuilder(Of ClosureDebugInfo).GetInstance()
+            Dim dynamicAnalysisSpans As ImmutableArray(Of SourceSpan) = ImmutableArray(Of SourceSpan).Empty
 
             body = Rewriter.LowerBodyOrInitializer(method,
                                                    methodOrdinal,
                                                    body,
                                                    previousSubmissionFields,
                                                    compilationState,
+                                                   CBool(_moduleBeingBuiltOpt?.EmitOptions.EmitDynamicAnalysisData),
```

Comment:
I believe this is evaluated as `CBool(DirectCast(Nothing, Boolean?))` which throws an `Exception`.